### DISCUSSION TITLE
Brew won't install arm64 version of bash

### DISCUSSION THREAD

**Discussion by petarkolaric** [Homebrew/discussions#2486] [Everyday usage] [answered]
Hi everyone!

I have just migrated to an Apple silicon mac. I used the Migration Assistant to do so. It correctly copied across homebrew, but I have not been able to use brew to install a native arm64 version of bash (which is the shell I usually use).

According to the formula there is an arm64 version:
https://formulae.brew.sh/formula/bash
https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/bash.rb#L68

When I check the file, it always reports as being the x86_64 architecture version:
```
petar@MacBook-Pro dev % file /usr/local/bin/bash    
/usr/local/bin/bash: Mach-O 64-bit executable x86_64
```

When I check my `brew config`, it incorrectly shows my operating system as the x86_64 version:
```
petar@MacBook-Pro dev % brew config                     
HOMEBREW_VERSION: 3.3.4
ORIGIN: https://github.com/Homebrew/brew
HEAD: 819eea5669807c1cf29e0d29985fc765ad7a9597
Last commit: 3 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: cae59a5099038038c79328be41a63cbf1aad274c
Core tap last commit: 73 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: 10-core 64-bit westmere
Clang: 12.0.0 build 1200
Git: 2.33.0 => /usr/local/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.0.1-x86_64
CLT: 12.0.0.32.29
Xcode: N/A
Rosetta 2: true
```

So a few questions:
- Is this something that should work? I was under the impression that brew was able to install arm64 formula now.
- How do I rebuild my config so it shows the correct OS?
- How do I install the arm64 version of bash using brew?

If I've missed anything obvious, sorry in advance.
Thanks for all your help everyone :)

**Comment by carlocab** [comment, marked as answer]
`/usr/local` installs of Homebrew run under Rosetta, so what you're seeing is expected. If you want to install native software, you need to install Homebrew into `/opt/homebrew`. You can use the install script to do this.

Make sure your terminal is not running under Rosetta (e.g. using `/usr/local/bin/bash`) when you run Homebrew's install script, and don't forget to follow the `Next Steps` about adding Homebrew to your `PATH`.

**Reply by petarkolaric** [reply]
Thanks for the help!

I am currently using zsh for my shell, and Hyper for my terminal. Both are running the AS version (I've checked in Activity Monitor).

I've just reinstalled brew, and made sure to follow the setup steps as a part of the script. It's showing that homebrew is installed under the /opt directory:
```
petar@MacBook-Pro dev % which brew
/opt/homebrew/bin/brew
```

I've tried to re-install bash (after making sure brew is in the right place) and whilst it says it's pouring the arm64 version, the new version remains as the x86_64 version:
```
petar@MacBook-Pro dev % brew install bash
==> Downloading https://ghcr.io/v2/homebrew/core/bash/manifests/5.1.8
Already downloaded: /Users/petar/Library/Caches/Homebrew/downloads/f899d2f6557e6c978bdc16d545707011cb09335854238c31d8c6bca3c65377c1--bash-5.1.8.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/bash/blobs/sha256:ca2a074414b936f14ae6736cbc15dfc773183ba45c4d1029d669473dae303e25
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:ca2a074414b936f14ae6736cbc15dfc773183ba45c4d1029d669473dae303e25?se=2021-11-
######################################################################## 100.0%
==> Pouring bash--5.1.8.arm64_monterey.bottle.tar.gz
🍺  /opt/homebrew/Cellar/bash/5.1.8: 157 files, 11.5MB
```
```
petar@MacBook-Pro dev % file /usr/local/bin/bash
/usr/local/bin/bash: Mach-O 64-bit executable x86_64
```

**Reply by carlocab** [reply]
Your ARM64 bash is in `/opt/homebrew/bin/bash`. Everything Homebrew installs into `/usr/local` is meant to run under Rosetta, so they will all be x86_64 binaries.

**Reply by petarkolaric** [reply]
Just checked - there is definitely an arm64 version of bash being installed under `/opt/homebrew/bin/bash`!

I've just run `brew link bash` (using the arm64 version of brew) and I now have the new arm64 version available when I run `which -a bash`.

Thanks so much for your help, I've been messing around with this for _ages_
